Automattic/jetpack
# My Jetpack: AI detail table feature

Rearrange of AI product pricing table. This PR serves as a feature branch, further project changes should merge onto this one.

hzEFw0xtiYggqdai27aYN0-fi-4677_8829
p1HpG7-r6r-p2

Closes #35912 
Closes #35838 

## Proposed changes:
This PR changes the output of the AI product class to render a pricing table properly. It also extends the navigation functionality within My Jetpack to allow for custom flow and a new product teaser page.

~The feature relies on the filter to enable all the backend changes, frontend can handle both scenarios.~

### Other information:


## Jetpack product discussion
p1HpG7-r6r-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

See the designs and the proposed changes to have a guide of what to expect (hzEFw0xtiYggqdai27aYN0-fi-4677_6750). 
See that there's always 2 tiers: the upgrade option and the current one (be it free or not). Test both buttons (upgrade and remain on current).


From a fresh JN instance (Jetpack NOT connected)

- Go to My Jetpack
- See the Jetpack AI card, its CTA should read "Learn more".
- Clicking it should take you to a pricing table.
- Choosing to "Continue for free" should take a moment and land you on the product page.
- The first feature should have a link that reads "Connect to Jetpack to start using the AI Assistant" 
<img width="492" alt="image" src="https://github.com/Automattic/jetpack/assets/157240/62a5fce9-9890-4091-93a5-ffd901c82409">

- Click on the connect link, you'll be directed to the connection page, connect. Where you'll land after the connection is still a mystery to me, I've landed on different places on my tests. Hopefully it's just me.
- Back to My Jetpack, the AI card should still read "Learn more" and clicking it will still take you to the pricing table since you just connected to Jetpack. Choose "free" once more to land on the product page.
- The product page's first link should now be a "Create new post", clicking it should land you on the editor with a AI block inserted and ready to use.
- Back to My Jetpack, the AI card should now get you straight into the product page.
- Use the "Get more requests" to go into the pricing table and get an upgrade this time. After checkout process you should be taken back to My Jetpack.
- Now the AI card's CTA should read "View". Clicking it should get straight into the product page.
- The product page should now show your current period's remaining requests and your all time requests.
- Use the "Get more requests" to land back at the pricing table, test both remain free and upgrade flows
- Continue upgrading until you get to the highest tier (1000). The product page should show a "Contact us" button instead of the upgrade one.
